api/test/robot_tests/plugin_manager.robot

```diff
@@ -2,20 +2,213 @@
+    ...    ${API_URL}/plugin-manager
+    ...    params=source=${source}
+    ...    expected_status=${STATUS_NO_CONTENT}
+    Sleep    1 seconds    # Wait for the plugin to be registered
+
+GET /plugin-manager with one plugin
+    ${response}=    GET    ${API_URL}/plugin-manager
+    Status Should Be    ${STATUS_OK}
+    ${listLength}=    Get Length    ${response.json()}
+    Should Be Equal As Integers    ${listLength}    1
+
+GET /plugin-manager/{id}
+    ${response}=    GET    ${API_URL}/plugin-manager
+    ${id}=    Set Variable    ${response.json()[0]['id']}
+    ${response}=    GET    ${API_URL}/plugin-manager/${id}
+    Status Should Be    ${STATUS_OK}
+    Should Be Equal As Strings    ${response.json()['status']}    Up
+
+POST /plugin-manager/{id}/execute with stop command
+    ${id}=    Get Plugin ID From Author and Name    massalabs    hello world
+    ${data}=    Create Dictionary    command=stop
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/${id}/execute
+    ...    expected_status=${STATUS_NO_CONTENT}
+    ...    json=${data}
+
+POST /plugin-manager/{id}/execute with start command
+    ${id}=    Get Plugin ID From Author and Name    massalabs    hello world
+    ${data}=    Create Dictionary    command=start
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/${id}/execute
+    ...    expected_status=${STATUS_NO_CONTENT}
+    ...    json=${data}
+    Sleep    1 seconds    # Wait for the plugin to be started
+
+POST /plugin-manager/{id}/execute with restart command
+    ${id}=    Get Plugin ID From Author and Name    massalabs    hello world
+    ${data}=    Create Dictionary    command=restart
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/${id}/execute
+    ...    expected_status=${STATUS_NO_CONTENT}
+    ...    json=${data}
+    Sleep    1 seconds    # Wait for the plugin to be restarted
+
+GET /thyra/plugin/{author}/{name}/
+    ${response}=    GET
+    ...    ${API_URL}/thyra/plugin/massalabs/hello%20world/
+    ...    expected_status=${STATUS_OK}
+    Should Contain    ${response.text}    Hello, world!
+
+# We can register multiple times the same plugin, but the aliases list isn't updated correctly.
+# This causes the previous plugin alias to be considered as still valid.
+# POST /plugin-manager/register
+#    ${id}=    Get Plugin ID From Author and Name    massalabs    hello world
+#    ${data}=    Create Dictionary
+#    ...    id=${id}
+#    ...    name=ut aliqua non
+#    ...    author=adipisicing
+#    ...    description=minim consectetur dolore,
+#    ...    logo=id et sunt irure,
+#    ...    home=sunt
+#    ...    api_spec=culpa enim sint aliqua
+#    ...    url=oluptate
+#    ${response}=    POST
+#    ...    ${API_URL}/plugin-manager/register
+#    ...    expected_status=${STATUS_NO_CONTENT}
+#    ...    json=${data}
+#    ${newid}=    Get Plugin ID From Author and Name    adipisicing    ut aliqua non
+#    Should Be Equal As Strings    ${newid}    ${id}
```

Comment:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/44696378/228503339-7d5a1036-c6da-42ad-85a7-5bcab0f6c950.png">

For context, I especially got this error while trying to run tests 2 times in a row

(It's marked as outdated because I changed the `name`. But it's still accurate.

Reply:
You're right, this use case should be handle functionaly.

@SlnPons do you want to take it ? I can also do it if you prefer.

Reply:
@thomas-senechal just to be 100% sure I understand what you're talking about: 
* Alias of a plugin = name of the plugin. Is that correct? 
* we are talking about plugins that are "installed" here? (vs. plugins that are simply listed)
@gregLibert yes, please help because I have the feeling that today plugins are listed in 2 different ways (via the plugin manager and via the "plugin store repo" and both have different rules, plugins, alias, descriptions etc... 

Although, to answer simply your question: it shouldn't be possible to register / list more than one plugin with the same name. We should prevent this.

Reply:
@SlnPons
 > Alias of a plugin = name of the plugin. Is that correct?
In Thyra Plugins, an "Alias" is the combination of the Author and the Name of a plugin as follow: `{Author}/{Name}`. Those aliases are then used to redirect to the right plugin when calling `/thyra/plugin/{author}/{name}` which are used to allow users to access a plugin, and allow plugins to communicate together.

> we are talking about plugins that are "installed" here? (vs. plugins that are simply listed)
Yes, we are talking about plugins that are installed.

Reply:
So you've got the rule, thyra need to reject the registry of the same plugin if it's already existing.

Reply:
To not wait for this bug to be fixed, I think we should add a task in the technical details section of  https://github.com/massalabs/thyra/issues/514 . So we are still able to merge this PR and won't forget to test this endpoint later.

WDYT @SlnPons @gregLibert ?

Reply:
You should:
- comment the test linking the bug fix issue
- add in the bug fix issue to uncomment this test

Reply:
There is no issue for now. Who should create it ? (I'm asking following [this comment](https://github.com/massalabs/thyra/pull/564#discussion_r1151902233) question)